core/server/data/migrations/versions/4.0/20-regenerate-posts-html.js

```diff
@@ -0,0 +1,75 @@
+const logging = require('../../../../../shared/logging');
+const {createIrreversibleMigration} = require('../../utils');
+const mobiledocLib = require('../../../../lib/mobiledoc');
+const htmlToText = require('html-to-text');
+
+module.exports = createIrreversibleMigration(async (knex) => {
+    logging.info('Starting re-generation of posts html.');
+
+    await knex.transaction(async (trx) => {
+        // get list of posts ids, use .forUpdate to lock rows until the transaction is finished
+        const postIdRows = await knex('posts')
+            .transacting(trx)
+            .forUpdate()
+            .select('id');
+
+        // transform each post individually to avoid dumping all posts into memory and
+        // pushing all queries into the query builder buffer in parallel
+        // https://stackoverflow.com/questions/54105280/how-to-loop-through-multi-line-sql-query-and-use-them-in-knex-transactions
+
+        for (const postIdRow of postIdRows) {
+            const {id} = postIdRow;
+            const [post] = await knex('posts')
+                .transacting(trx)
+                .where({id})
+                .select('mobiledoc');
+
+            let mobiledoc;
+
+            try {
+                mobiledoc = JSON.parse(post.mobiledoc || null);
+
+                if (!mobiledoc) {
+                    return logging.warn(`No mobiledoc for ${post.id}. Skipping.`);
+                }
+            } catch (err) {
+                return logging.warn(`Invalid JSON structure for ${post.id}. Skipping`);
+            }
+
+            const html = mobiledocLib.mobiledocHtmlRenderer.render(mobiledoc);
+
+            const updatedAttrs = {
+                html: html
+            };
+
+            // NOTE: block comes straight from the Post model
+            // https://github.com/TryGhost/Ghost/blob/4.0.0-alpha.2/core/server/models/post.js#L484
+            if (html !== post.html || !post.plaintext) {
```

Comment:
You're trying to access `post.html` and `post.plaintext` here but it's not there in the select at [line 25](https://github.com/TryGhost/Ghost/pull/12660/files#diff-13dab3d53b700bbe52a3246066a4ba53f513c23d9d3f09cdd54e71101b8c4b73R25). I'd add these columns to the select.

Reply:
Good catch! 🎣 Force-pushed with a fix, thanks!